aspnetcore/fundamentals/change-tokens.md

```diff
@@ -2,169 +2,187 @@
+The sample app demonstrates two implementations for monitoring configuration changes. If any of the *appsettings* files change, both of the file monitoring implementations execute custom code&mdash;the sample app writes a message to the console.
 
-A configuration file's `FileSystemWatcher` can trigger multiple token callbacks for a single configuration file change. The sample's implementation guards against this problem by checking file hashes on the configuration files. Checking file hashes ensures that at least one of the configuration files has changed before running the custom code. The sample uses SHA1 file hashing (*Utilities/Utilities.cs*):
+A configuration file's `FileSystemWatcher` can trigger multiple token callbacks for a single configuration file change. To ensure that the custom code is only run once when multiple token callbacks are triggered, the sample's implementation checks file hashes. The sample uses SHA1 file hashing. A retry is implemented with an exponential back-off. The retry is present because file locking may occur that temporarily prevents computing a new hash on a file.
 
-   [!code-csharp[](change-tokens/sample/Utilities/Utilities.cs?name=snippet1)]
+*Utilities/Utilities.cs*:
 
-   A retry is implemented with an exponential back-off. The re-try is present because file locking may occur that temporarily prevents computing a new hash on one of the files.
+[!code-csharp[](change-tokens/samples/2.x/SampleApp/Utilities/Utilities.cs?name=snippet1)]
 
 ### Simple startup change token
 
-Register a token consumer `Action` callback for change notifications to the configuration reload token (*Startup.cs*):
+Register a token consumer `Action` callback for change notifications to the configuration reload token.
+
+In `Startup.Configure`:
 
-[!code-csharp[](change-tokens/sample/Startup.cs?name=snippet2)]
+[!code-csharp[](change-tokens/samples/2.x/SampleApp/Startup.cs?name=snippet2)]
 
 `config.GetReloadToken()` provides the token. The callback is the `InvokeChanged` method:
 
-[!code-csharp[](change-tokens/sample/Startup.cs?name=snippet3)]
+[!code-csharp[](change-tokens/samples/2.x/SampleApp/Startup.cs?name=snippet3)]
 
-The `state` of the callback is used to pass in the `IHostingEnvironment`. This is useful to determine the correct *appsettings* configuration JSON file to monitor, *appsettings.&lt;Environment&gt;.json*. File hashes are used to prevent the `WriteConsole` statement from running multiple times due to multiple token callbacks when the configuration file has only changed once.
+The `state` of the callback is used to pass in the `IHostingEnvironment`. This is useful to determine the correct *appsettings* configuration JSON file to monitor, *appsettings.{ENVIRONMENT}.json*. File hashes are used to prevent the `WriteConsole` statement from running multiple times due to multiple token callbacks when the configuration file has only changed once.
```

Comment:
This 2nd sentence reads poorly. Seems like something is needed after "monitor, ".